CuddleTreeDefenseGame/Assets/Scripts/ControlHandler.cs

```diff
@@ -1,16 +1,43 @@
 using UnityEngine;
+using System.Collections;
 
 public class ControlHandler : MonoBehaviour
 {
     [SerializeField] GameObject buildPrefabWithButtonT;
+    private bool isTPressed;
+    private GameObject obj;
+    private Coroutine myCoroutine;
+
     void Update()
     {
         if (Input.GetKeyDown(KeyCode.T))
+        {
+            isTPressed = true;
+            obj = Instantiate(buildPrefabWithButtonT);
+            Tools.SetColorOnGameObject(obj, new Color(1, 1, 1, 0.25f));
+            Tools.ToggleScriptsInGameObject(obj, false);
+            myCoroutine = StartCoroutine(FollowMouse(obj));
+        }
+        if ( isTPressed && Input.GetMouseButton(0))
+        {
+            isTPressed = false;
+            StopCoroutine(myCoroutine);
+            Tools.SetColorOnGameObject(obj, new Color(1, 1, 1, 1));
+            Tools.ToggleScriptsInGameObject(obj, true);
+        }
+
+    }
+    IEnumerator FollowMouse(GameObject src)
+    {
+        while (true)
         {
             Vector3 worldpos = Camera.main.ScreenToWorldPoint(Input.mousePosition);
```

Comment:
Vi borde prata om C# code conventions, eftersom för tillfället gör alla olika.
Enligt .NET convention skall alla classer och metoder använda pascal case (https://www.dofactory.com/reference/csharp-coding-standards#classnames)

Samt ska `var` användas för att deklarera -_lokala_- variabler, förutom primitva datatyper som t.ex.: int, bool och string. (Se https://www.dofactory.com/reference/csharp-coding-standards#typenames)
Anledningen dom skrev: `Why: removes clutter, particularly with complex generic types. Type is easily detected with Visual Studio tooltips.`

Om vi inte ska använda conventions får vi komma överens om det, så att alla gör samma.